tictactoe.js

```diff
@@ -0,0 +1,144 @@
+const readline = require("readline");
+
+const readlineInterface = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Ask question
+const ask = questionText => {
+  return new Promise((resolve, reject) => {
+    readlineInterface.question(questionText, resolve);
+  });
+};
+
+let values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+let currentPlayer = "X";
+let winner = "";
+
+const grid = gridValues => {
+  return (
+    "\n" +
+    " " +
+    gridValues[1] +
+    " | " +
+    gridValues[2] +
+    " | " +
+    gridValues[3] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[4] +
+    " | " +
+    gridValues[5] +
+    " | " +
+    gridValues[6] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[7] +
+    " | " +
+    gridValues[8] +
+    " | " +
+    gridValues[9] +
+    "\n\n" +
+    "Player " +
+    currentPlayer +
+    "'s turn\n" +
+    "Move to?\n"
+  );
+};
+
+const checkWinner = player => {
+  if (
+    (values[1] === player && values[2] === player && values[3] === player) ||
+    (values[4] === player && values[5] === player && values[6] === player) ||
+    (values[7] === player && values[8] === player && values[9] === player) ||
+    (values[1] === player && values[4] === player && values[7] === player) ||
+    (values[2] === player && values[5] === player && values[8] === player) ||
+    (values[3] === player && values[6] === player && values[9] === player) ||
+    (values[1] === player && values[5] === player && values[9] === player) ||
+    (values[3] === player && values[5] === player && values[7] === player)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+start();
+
+async function start() {
+  let prompt = "start";
+  let spotTaken = false;
+
+  while (prompt.toLowerCase() !== "exit") {
+    if (spotTaken) {
+      prompt = await ask(
+        "\nSorry spot is already taken, try again!! \n\n" + grid(values)
+      );
+    } else {
+      prompt = await ask(grid(values));
+    }
+
+    spotTaken = false;
+    if (prompt === "1" && values[1] === 1) {
```

Comment:
this code is very duplicative... try refactoring it. In VS Code, select all lines from 82-103, right click, and select "Refactoring" then "Extract Function" to start with.The function should continue to return a boolean `spotTaken` for use later on. Then see if you can write it as a loop instead of a series of `else if`s.